File: record.cpp
#include <iostream>
#include "freewili_api.h"  // Include the actual API header if available


void recordAudio() {
    std::string filename = "test_audio.wav";
    int duration = 5000; // 5 seconds

    std::cout << "Recording audio for " << duration / 1000 << " seconds...\n";
    
    recordSound(filename.c_str(), duration);

    std::cout << "Recording complete. Saved as " << filename << "\n";
}

void playRecordedAudio() {
    std::string filename = "test_audio.wav";
    
    std::cout << "Playing recorded audio: " << filename << "\n";
    playSound(filename.c_str());
}


int main() {
    recordAudio();           // Record for 5 seconds
    playRecordedAudio();

    return 0;
}

File: record.js
// Step 1: Record 5 seconds of audio
recordSound("my_audio.wav", 5000);

// Step 2: Enable streaming
setAudioSettings({
  stream: true,
  fft: true,
  mic: true
});

// Step 3: Detect loud sounds
if (getMicLevel() > 80) {
  console.log("Loud sound detected!");
}

// Step 4: Play back the recorded audio
setTimeout(() => {
  playSound("my_audio.wav");
}, 6000); // Delay playback to ensure recording is finished
